### PR TITLE
Added support for bitbucket repositories

### PR DESCRIPTION
**Pull request by pgorgon-hem**
* added variable BITBUCKET_REPOSITORY
* added short syntax in form "bb:user/repo"
* added description of that to readme.md
Connected feature request: #255

### CODE REVIEW

**Comment by TheLartians** [comment]
Hey, thanks for the PR! Could you add a test case for the shorthand syntax in [`test/unit/parse_add_package_single_arg.cmake`](https://github.com/cpm-cmake/CPM.cmake/blob/master/test/unit/parse_add_package_single_arg.cmake)?

Also it looks like CMake.org's SSL certificate has expired, which is why some of the test cases are failing. I'll check and relaunch the remaining tests later.

**Comment by pgorgon-hem** [comment]
Hi! Of course I will add example and test. In case of cmake.org SSL certificate there is probably no choice and we have to wait for renew of cert.

**Review by iboB — DISMISSED** [review]
Thanks

File: cmake/CPM.cmake [hunk 1]
```diff
@@ -405,6 +409,10 @@ function(CPMAddPackage)
     set(CPM_ARGS_GIT_REPOSITORY "https://gitlab.com/${CPM_ARGS_GITLAB_REPOSITORY}.git")
   endif()
 
+  if(DEFINED CPM_ARGS_BITBUCKET_REPOSITORY)
+    set(CPM_ARGS_GIT_REPOSITORY "https://bitbucket.org/${CPM_ARGS_BITBUCKET_REPOSITORY}.git")
+  endif()
```
Comment (TheLartians) [inline cmake/CPM.cmake:414]: Perhaps we could group the three `if` clauses using `elseif` as well.

**Review by TheLartians — CHANGES_REQUESTED** [review]
@pgorgon-hem thanks for adding the test! Could you still add a short example using a bitbucket repo for integration testing?

File: cmake/CPM.cmake (hunk 1 above)
Comment (pgorgon-hem) [reply cmake/CPM.cmake]: I grouped it and pushed.

**Review by TheLartians — DISMISSED** [review]
Thanks for the changes, looks great! Just to be sure, are you still planning on adding an example?

**Comment by pgorgon-hem** [comment]
I added example repo on BB - our TestingFramework, but I'm not sure whether it compiles on server because of minimum cmake version required in our code.

**Comment by TheLartians** [comment]
Yep, this definitely breaks CI, as it requires CMake 3.19 (we are still on 3.14). 

<details><summary>Error message</summary>
<p>

```
running example TestingFramework
1263================================
1264- cmake -Hexamples/TestingFramework -Bbuild/TestingFramework
1265error while running 'cmake -Hexamples/TestingFramework -Bbuild/TestingFramework':
1266   CMake Error at /home/travis/build/cpm-cmake/CPM.cmake/build/TestingFramework/_deps/testingframework-src/CMakeLists.txt:1 (cmake_minimum_required):
1267    CMake 3.19 or higher is required.  You are running version 3.14.0
1268
```

</p>
</details>

I've updated CMake to 3.20 in #264, and will let you know once you can merge!

**Comment by pgorgon-hem** [comment]
Great, thanks! So I'm waiting for new cmake.

**Comment by iboB** [comment]
The style still needs to be fixed, right?

Otherwise this works with #264 in

**Comment by pgorgon-hem** [comment]
Now everything should be ok including styles (I tested it on our fork).

**Review by TheLartians — APPROVED** [review]
Looks great, thanks for adding the test and example!

**Comment by TheLartians** [comment]
The changes are now published in `v0.32.2`! 🎉

**Pull request merged**